distributed/worker_memory.py

```diff
@@ -302,32 +313,46 @@ def __init__(
             dask.config.get("distributed.worker.memory.monitor-interval"),
             default=None,
         )
+        self.nanny = weakref.ref(nanny)
```

Comment:
This is self._worker above - could you make it consistent?
Also, please add to the attributes declarations above in both classes.